pages/1_13/Key_Mouse_Input.md

```diff
@@ -0,0 +1,120 @@
+
+GLFW has a lot of enums set up for you to use in conjunction with the above two methods. The most important ones for mouse input are:
+
+`GLFW_MOUSE_BUTTON_LEFT`  
+`GLFW_MOUSE_BUTTON_RIGHT`  
+`GLFW_MOUSE_BUTTON_MIDDLE`
+
+In the following code snippet I'll show you how you can use them with the event parameter to check which key was pressed:
+
+```java
+    @SubscribeEvent
+    public static void onMouseClick(InputEvent.MouseInputEvent event){
+        if (event.getButton() == GLFW_MOUSE_BUTTON_LEFT){
+            //Do your stuff here
+        }
+    }
+```
+It's as simple as that! Whith that if statement we check which mouse button was pressed before we execute our desired code.  
+
+Now if we also want to check whether the mouse button was pressed or released we can add an extra check in the if statement using `event.getAction()`. This function also returns an int and you guessed it, GLFW also has a couple of enums for this:
+
+`GLFW_PRESS`  
+`GLFW_RELEASE`  
+
+We can just use them in the same if statement like this:
+
+```java
+    @SubscribeEvent
+    public static void onMouseClick(InputEvent.MouseInputEvent event){
+        if (event.getButton() == GLFW_MOUSE_BUTTON_LEFT && event.getAction() == GLFW_PRESS){
+            //Do your stuff here
+        }
+    }
+```
+This method checks if the **left** mouse button was **pressed**.
+
+That's it for mouse input! Let's continue to key input, which is very similar.
+
+### Key Input
+
+We already have all the imports we need from the previous section.  
+
+For our key input we're going to make another public static method with return type void and annotate it with `@SubscribeEvent`. Instead of passing in `InputEvent.MouseInputEvent event`, we're now going to pass in `InputEvent.KeyInputEvent event` as our method parameter. Your method should look like this:
+
+```java
+    @SubscribeEvent
+    public static void onKeyPress(InputEvent.KeyInputEvent event){
+
+    }
+```
+
+Now the event parameter for key input uses `event.getKey()` instead of `event.getButton()`. The only thing that changes is the key enums. They are constructed like this: `GLFW_KEY_R`. For all the keys that are available you should take a look inside the GLFW class, they're all there. Below you will find an example of a method that checks if the **R** key was **released**:
+
+```java
+    @SubscribeEvent
+    public static void onKeyRelease(InputEvent.KeyInputEvent event){
+        if (event.getKey() == GLFW_KEY_R && event.getAction() == GLFW_RELEASE){
+            //Do your stuff here
+        }
+    }
+```
+
+And that's basically all there's to it! You can make it as fancy as you want with multiple listeners, for example, `onKeyDown(), onKeyUP(), onKey()`. You can also use a switch case to check for more keys or even create a custom keybinding system.
```

Comment:
You mention a "custom keybind system", although I'm wondering if it's better to give an example of an actual vanilla custom keybinding? If you don't know how to, then I might try add it.

Reply:
I don't exactly know how to do it yet, I just know it's possible. Maybe for now we could just remove that suggestion?

Reply:
Yeah sure, just so not to confuse things. I'll possibly add the keybinds in myself later.